### PR TITLE
Add my-credit-rating-agency and Kamelet

### LINKED ISSUE #14: Provide a Customer Credit Rating integration example
### Description of Problem:
When integrating existing services with a given BIAN service domain, it is usually the case where each service is implemented using a different API.

Integration examples including Camel routes and message transformation will help shaping the interoperability needs for the Mercury project.

### Potential Solutions:
Camel K can definitely help with the deployment and to simplify the routing definitions. Besides, the Kamelet concept sounds promising in terms of providing an abstracted entrypoint regardless of the transport

### PR DESCRIPTION
**Pull request by ruromero**
Fix #14 

Signed-off-by: ruromero <rromerom@redhat.com>

Adding customer-credit-rating example with:

* Microservice for my-credit-rating-agency to be used as internal reporting
* Kamel Routes for both internal and external reporting
* Pipeline for building the my-credit-rating-agency
* Kamelet with basic CustomerCreditRating processing to CloudEvent
* BIAN Rest Utils project to be used in the Kamelet
* Camel-K route examples

### CODE REVIEW

File: examples/customer-credit-rating/deploy/quarkus-native-build-task.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      script: |
+        #!/usr/bin/env bash
+        root=$(echo "$(params.SUBDIRECTORY)" | cut -d/ -f1)
+        if [ -d "src" ]
+        then
+          rm -rf src pom.xml target
+        fi
+        mv -f $(params.SUBDIRECTORY)/$(params.SUBDIRECTORY)/* .
+        rm -rf $root
+        if [ ! -d "/home/kogito/.m2/repository" ]
```
Comment (ilan-pinto) [inline examples/customer-credit-rating/deploy/quarkus-native-build-task.yaml:25]: @ruromero why are we creating the kogito folder?

File: examples/customer-credit-rating/deploy/quarkus-native-build-task.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        then
+          mkdir /home/kogito/.m2/repository
+        fi
+      workingDir: $(workspaces.source.path)
+    - name: compile-quarkus-native-binary
+      image: quay.io/kiegroup/kogito-builder:1.7
```
Comment (ilan-pinto) [inline examples/customer-credit-rating/deploy/quarkus-native-build-task.yaml:31]: @ruromero  why are we using the kogito-builder?

File: examples/customer-credit-rating/my-credit-rating-agency/src/main/java/com/redhat/mercury/ccr/demo/CreditRatingResource.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                return ratings.get(tin);
+            }
+            throw new NotFoundException(tin);
+        });
+    }
+
+    /**
+     * Updates an existing `CreditRating`.
+     */
+    @Path("{tin}")
```
Comment (ilan-pinto) [inline examples/customer-credit-rating/my-credit-rating-agency/src/main/java/com/redhat/mercury/ccr/demo/CreditRatingResource.java:62]: @ruromero what does tin refers to? what is the meaning of the variable name?

Comment (ilan-pinto) [reply examples/customer-credit-rating/my-credit-rating-agency/src/main/java/com/redhat/mercury/ccr/demo/CreditRatingResource.java:62]: ok, i guess it's Tax Identification Number, suggesting to change to use full name and not initials

File: examples/customer-credit-rating/deploy/quarkus-native-build-task.yaml (hunk 1 above)
Comment (ruromero) [reply examples/customer-credit-rating/deploy/quarkus-native-build-task.yaml:25]: In the pipeline I decided to use the Kogito builder for simplicity when building a Quarkus project.
We can review it later. I just wanted something that worked and I'm more used to Kogito builds.

File: examples/customer-credit-rating/deploy/quarkus-native-build-task.yaml (hunk 2 above)
Comment (ruromero) [reply examples/customer-credit-rating/deploy/quarkus-native-build-task.yaml:31]: Related to the above comment

File: examples/customer-credit-rating/my-credit-rating-agency/src/main/java/com/redhat/mercury/ccr/demo/CreditRatingResource.java (hunk 3 above)
Comment (ruromero) [reply examples/customer-credit-rating/my-credit-rating-agency/src/main/java/com/redhat/mercury/ccr/demo/CreditRatingResource.java:62]: Yes. I consider _tin_ meaningful enough in this domain context and as it is an example I'll rather keep it as is

Comment (ilan-pinto) [reply examples/customer-credit-rating/my-credit-rating-agency/src/main/java/com/redhat/mercury/ccr/demo/CreditRatingResource.java:62]: If it is a common term then of course

**Pull request merged**